src/query/service/src/interpreters/interpreter_refresh_table_cache.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use databend_common_exception::ErrorCode;
+use databend_common_exception::Result;
+use databend_common_sql::plans::RefreshTableCachePlan;
+
+use super::Interpreter;
+use crate::pipelines::PipelineBuildResult;
+use crate::sessions::QueryContext;
+use crate::sessions::TableContext;
+
+pub struct RefreshTableCacheInterpreter {
+    ctx: Arc<QueryContext>,
+    plan: RefreshTableCachePlan,
+}
+
+impl RefreshTableCacheInterpreter {
+    pub fn try_create(ctx: Arc<QueryContext>, plan: RefreshTableCachePlan) -> Result<Self> {
+        Ok(RefreshTableCacheInterpreter { ctx, plan })
+    }
+}
+
+#[async_trait::async_trait]
+impl Interpreter for RefreshTableCacheInterpreter {
+    fn name(&self) -> &str {
+        "RefreshTableCacheInterpreter"
+    }
+
+    fn is_ddl(&self) -> bool {
+        true
+    }
+
+    #[async_backtrace::framed]
+    async fn execute2(&self) -> Result<PipelineBuildResult> {
+        let plan = &self.plan;
+        let catalog = self.ctx.get_catalog(&plan.catalog).await?;
+
+        if !catalog.support_partition() {
+            return Err(ErrorCode::TableEngineNotSupported(
```

Comment:
refresh_cache has nothing to do with `support_partition `.

You can throw the error in `refresh`  function.